src/result.test.ts

```diff
@@ -1235,6 +1376,22 @@ describe("Result", () => {
 				Result.assertError(nextResult);
 				expect(spy).not.toHaveBeenCalled();
 			});
+
+			it("transforms thrown errors when an errorTransform function is defined", async () => {
+				const result: Result<number, CustomError> = await Result.ok(
+					2,
+				).mapCatching(
+					async (): Promise<number> => {
+						throw new Error("TEST_ERROR");
+					},
+					(_error) => {
+						return new CustomError();
```

Comment:
This is an example of where `CustomError` creates a false positive.
If you replace this with `ErrorA` it will probably fail the typecheck. See comment about types of `mapCatching` in `result.ts`.